.github/workflows/deploy.yml

```diff
@@ -53,11 +66,36 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
-      - name: Build and push the image
+      - name: Set current date as env variable
+        run: echo "TODAY=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+      - name: Build and push the image (future)
         uses: docker/build-push-action@v3
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main' 
         with:
-          push: ${{ github.ref == 'refs/heads/main' || github.ref == 'refs/heads/future' }}
+          push: true
           platforms: linux/amd64,linux/arm64
           tags: |
-            ${{ secrets.PUBLIC_RUNNER_TERRAFORM_ECR_REPOSITORY_URL }}:${{ github.ref == 'refs/heads/main' && 'latest' || 'future' }}
-            ghcr.io/spacelift-io/runner-terraform:${{ github.ref == 'refs/heads/main' && 'latest' || 'future' }}
+            ${{ secrets.PUBLIC_RUNNER_TERRAFORM_ECR_REPOSITORY_URL }}:future
+            ghcr.io/spacelift-io/runner-terraform:future
+
+      - name: Build and push the image (latest)
+        uses: docker/build-push-action@v3
+        if: github.event_name == 'schedule' || startsWith(github.ref, 'refs/tags/v')
+        with:
+          push: true
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ${{ secrets.PUBLIC_RUNNER_TERRAFORM_ECR_REPOSITORY_URL }}:latest
+            ${{ secrets.PUBLIC_RUNNER_TERRAFORM_ECR_REPOSITORY_URL }}:${{ steps.latest-tag.outputs.tag }}_${{ env.TODAY }}
+            ghcr.io/spacelift-io/runner-terraform:latest
+            ghcr.io/spacelift-io/runner-terraform:${{ steps.latest-tag.outputs.tag }}_${{ env.TODAY }}
```

Comment:
Should we maybe use a `+` rather than `_`. I'm not sure `_` is valid in SemVer. `+` is used for build metadata: https://semver.org/#spec-item-10

The downside to that is it says that build metadata is ignored when ordering versions, but maybe that's not a problem?

Reply:
Initially it was a +, but + sign is not a valid docker tag :( Maybe dash?